### DISCUSSION TITLE
intersphinx.external usage and documentation questions

### DISCUSSION THREAD

**Discussion by jdillard** [sphinx-doc/sphinx#12937] [Q&A] [answered]
I recently discovered the `intersphinx.external` warning type for [suppress_warnings](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-suppress_warnings) through https://github.com/sphinx-doc/sphinx/issues/12585#issuecomment-2228420035, but it is unclear to me all the types of warnings `intersphinx.external` applies to. The [test_intersphinx_role](https://github.com/sphinx-doc/sphinx/blob/341b0417d5ccc2b495276cd60c5c6d8accd2a72c/tests/test_extensions/test_ext_intersphinx.py#L712-L721) test does seem to hint at some possibilities.

I was hoping to use `intersphinx.external` to suppress warnings where a reference isn't found in an external inventory, but that doesn't seem to be supported in my testing. So my question is, is it possible to use `intersphinx.external` to suppress warnings where a reference isn't found in an external inventory?

----

Also, in doing research, I was reading about the new(er) intersphinx syntax in the [sphinx.ext.intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html) docs and noticed a potential error on [L81](https://github.com/sphinx-doc/sphinx/blob/341b0417d5ccc2b495276cd60c5c6d8accd2a72c/doc/usage/extensions/intersphinx.rst?plain=1#L81). It has a typo with a missing leading colon, but also the syntax doesn't match the syntax in the [external role](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#role-external) docs, making the proper syntax:
```
:external+python:ref:`comparison manual <comparisons>`
```

I'm happy to make a PR to fix the typo, but didn't know if more (small, as I know there is a lot of sweeping improvements to be made) docs updates would come out of a potential discussion.

edit: PR made: https://github.com/sphinx-doc/sphinx/pull/12941

**Comment by jdillard** [comment, marked as answer]
Ahhh, I should be using `ref.ref`, `ref.doc`, etc. I'll leave this up if anyone has additional docs improvement suggestions, otherwise I'll give it some thought and try based on my attempt at using suppress_warnings for such things.

**Reply by jayaddison** [reply]
Hey @jdillard - I noticed your documentation fixup pull request related to this, thanks!

I didn't quite understand what you mean by using `ref.ref` / etc instead, though - what do those affect?

**Reply by jdillard** [reply]
I'm away from my computer so don't have links to code handy, but essentially when the interspinx extension handles reference warnings it assigns the type.subtype as ref.ref, ref.doc, etc. So that means you need to use those to suppress warnings.  I personally think it would be better if the type.subtype was something like ref.external+ref, ref.external+doc, etc for external references so there was more granular control of warning suppression.
